api.go

```diff
@@ -0,0 +1,122 @@
+package gvm
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// GoRelease represents a Go release from the go.dev API
+type GoRelease struct {
+	Version string   `json:"version"`
+	Stable  bool     `json:"stable"`
+	Files   []GoFile `json:"files"`
+}
+
+// GoFile represents a downloadable file for a Go release
+type GoFile struct {
+	Filename string `json:"filename"`
+	OS       string `json:"os"`
+	Arch     string `json:"arch"`
+	Version  string `json:"version"`
+	SHA256   string `json:"sha256"`
+	Size     int64  `json:"size"`
+	Kind     string `json:"kind"`
+}
+
+// fetchGoReleases fetches the list of Go releases from the go.dev API
+func (m *Manager) fetchGoReleases() ([]GoRelease, error) {
+	apiURL := "https://go.dev/dl/?mode=json&include=all"
+
+	client := &http.Client{
+		Timeout: m.HTTPTimeout,
+	}
+
+	resp, err := client.Get(apiURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch Go releases: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("API request failed with status %d", resp.StatusCode)
+	}
+
+	var releases []GoRelease
+	if err := json.NewDecoder(resp.Body).Decode(&releases); err != nil {
+		return nil, fmt.Errorf("failed to decode API response: %w", err)
+	}
+
+	return releases, nil
+}
+
+// findArchiveFile finds the archive file for the given OS/arch combination
+func (r *GoRelease) findArchiveFile(goos, goarch string) *GoFile {
+	// Map special arch cases
+	archToMatch := goarch
+	if goos == "linux" && goarch == "armv6l" {
```

Comment:
This looks wrong. The check should be if GOARCH is "arm" then change the architecture to be armv6l (which is the only variant that binaries are released for.